mako/cmd.py

```diff
@@ -51,13 +51,33 @@ def cmdline(argv=None):
         default=None,
         help="Write to file upon successful render instead of stdout",
     )
-    parser.add_argument("input", nargs="?", default="-")
+    parser.add_argument(
+        "--strip-shebang", "-s", action="store_true",
+        help="Strip a shebang in the first line of given input file",
+    )
+    parser.add_argument(
+        "--shift-args", "-a", action="store_true",
+        help="pass further `args` of this argument parser to the template"
+        "as sys.argv - allowing the template to do argparsing again.",
+    )
+    parser.add_argument("input", nargs="?", default="-",
+                        help="file to parse as template. default is stdin.")
+    parser.add_argument("args", nargs="*",
+                        help=("arguments passed to template's script. "
+                              "use '-- args...' when first arg "
+                              "starts with - or "))
 
     options = parser.parse_args(argv)
 
     output_encoding = options.output_encoding
     output_file = options.output_file
 
+    # strip away mako's original args, so the template can get new args
+    # we do this so templates can handle further args on their own
+    if options.shift_args:
+        saved_sys_argv = sys.argv
+        sys.argv = [options.input] + options.args
```

Comment:
hey there -

what I want is:

1. do not change sys.argv at all
2. make a new template variable "sys_argv" with `[options.input] + options.args`, or whatever is most appropriate for argparse
3. users that want to use argparse in a template use `parser.parse_args(sys_argv)`

that's it!

Reply:
okay great, I'll implement that :)

Reply:
I'll mark it resolved when the next patch is put up (with the issue resolved :) )